lib/Cake/View/Helper/HtmlHelper.php

```diff
@@ -829,6 +833,19 @@ public function image($path, $options = array()) {
 			unset($options['url']);
 		}
 
+		if (!empty($options['base64']) && !$url) {
+			$fullPath = WWW_ROOT . $path;
+			$file = new File($fullPath, false);
+			if (!$file->exists()) {
+				throw new InvalidArgumentException(__d('cake', 'Unable to find the requested image to output as base64!'));
+			}
+			$file->open();
```

Comment:
This should go above along with the `$file->exits()` check. A file might exist but might not be readable due to permissions issue.